### PR TITLE
Stable default behaviour

### PR DESCRIPTION
**Pull request by OliverZim**
This pull request includes the discussed changes to user management in the Dockerfile, as well as default behavior when no GPU devices are chosen and no caching directory is provided. The Readme is updated accordingly.

### CODE REVIEW

File: .vscode/settings.json
Comment (konstantinjdobler) [inline .vscode/settings.json:1]: We can keep this as default settings for the linter & formatter in VS Code

File: scripts/console.sh [hunk 1]
```diff
@@ -11,14 +11,25 @@ if [ "$current_directory" == "scripts" ]; then
...
+# if you plan on mounting a cache-folder you will have to create one, before you can run this script
+CACHE_DIR="NONE"
+
+# Change the following image-tag to the name of your own image, if you do not want to use the default one
+IMAGE_TAG="konstantinjdobler/nlp-research-template"
+
 docker run -it \
-    --gpus='device=CHANGE_ME' \
+    --user $(id -u):$(id -g) \
+    $([[ "$DEVICES" != "NONE" ]] && echo "--gpus=\"device=$DEVICES\"") \
```
Comment (konstantinjdobler) [inline scripts/console.sh:26]:
```suggestion
    $([[ "$DEVICES" != "NONE" ]] && echo "--gpus device=$DEVICES") \
```

Did this work before like it is in the PR currently/ According to the docs, you have to use `--gpus='"device=$DEVICES"'` or `--gpus device=$DEVICES`

Comment (OliverZim) [reply scripts/console.sh:26]: It is working this way. I tested selecting only one, multiple or no GPUs.

File: train.py [hunk 2]
```diff
@@ -401,7 +401,7 @@ def main(parsed_arg_groups: tuple[TrainingArgs, MiscArgs]):
         max_steps=args.training_goal,
         val_check_interval=args.val_frequency,
         check_val_every_n_epoch=None,  # validation based on steps instead of epochs
-        devices=args.devices,
+        devices=num_devices,    # determine the number of devices that are actually there (use one CPU if no GPUs are available)
```
Comment (konstantinjdobler) [inline train.py:404]: This breaks specifying `--devices=0,4,5` explicitly. But let's just make a note of it and I will fix it when I refactor the entire device handling in #32

**Review by konstantinjdobler — APPROVED** [review]
Looks good!

**Pull request merged**